docs/concepts/configuration/overview.md

```diff
@@ -45,7 +45,7 @@ This is a living document. If you think of something that is not on this list bu
     2. Create a service;
     3. Then scale up the replication controller.
 
-- Don't use `hostPort` unless it is absolutely necessary (for example: for a node daemon). It specifies the port number to expose on the host. When you bind a Pod to a `hostPort`, there are a limited number of places to schedule a pod due to port conflicts— you can only schedule as many such Pods as there are nodes in your Kubernetes cluster.
+- Don't use `hostPort` unless it is absolutely necessary (for example: for a node daemon). It specifies the port number to expose on the host.  When you bind a Pod to a `hostPort`, there are a limited number of places to schedule a pod due to port conflicts. The conflict comes from the same `hostIP:hostPort:protocol` combination, different `hostIP:hostPort:protocol` combinations means different requirements. For example, a pod which binds tcp 80 host port on 127.0.0.1 does not conflict with another pod which binds tcp 80 port on 127.0.0.2. Special notes on hostIP and protocol, if you don't specify the hostIP and protocol, kubernetes will give you 0.0.0.0 and tcp as default hostIP and protocol.
 
```

Comment:
The conflict comes from being limited to a single `hostIP:hostPort:protocol` combination. For example, a pod that binds tcp 80 host port on 127.0.0.1 does not conflict with another pod that binds tcp 80 port on 127.0.0.2.
Note that if you don't specify the hostIP and protocol, Kubernetes gives you 0.0.0.0 and tcp as the default hostIP and protocol.

Reply:
good to know @steveperry-53

Reply:
I would use language like this (insert after ...there are a limited number of places to schedule a pod due to port conflicts.)

A pod with a hostPort can only be scheduled to a host if the network endpoint three-tuple <hostIP,hostPort,Protocol> is unique, i.e. there is no other Pod or process in the cluster with the same network point.

Reply:
@bowei I don't think so. what if pod wants to use 0.0.0.0/80/TCP, the cluster can have as many such pod as  nodes in cluster. the sentence "there is no other Pod or process in the cluster with the same network endpoint."  "in the cluser" is too wide.

Reply:
Add `"0.0.0.0" is a wildcard hostIP that will occupy all <_,hostPort,protocol> on the node the pod is scheduled on. Specifically, it will occupy all <IP,hostPort,protocol> for all IPs on the host.`

Reply:
My general comment is that the documentation should explain exactly when a conflict will occur so the user does not have to go somewhere else to find the information.

Reply:
>My general comment is that the documentation should explain exactly when a conflict will occur so the user does not have to go somewhere else to find the information.

I can't agree more